src/hooks/useBalances.ts

```diff
@@ -0,0 +1,39 @@
+import { mIOToken } from '@ar.io/sdk/web';
+import { AR } from '@src/constants';
+import { useGlobalState } from '@src/store';
+import { ArweaveTransactionID } from '@src/utils/ArweaveTransactionId';
+import { useQuery } from '@tanstack/react-query';
+
+export type Balances = { ar: number; io: number };
+
+const useBalances = (walletAddress?: ArweaveTransactionID) => {
+  const arIOReadSDK = useGlobalState((state) => state.arIOReadSDK);
+  const arweave = useGlobalState((state) => state.arweave);
+  const blockHeight = useGlobalState((state) => state.blockHeight);
+
+  const res = useQuery<Balances>({
+    queryKey: ['balances', arIOReadSDK, arweave, blockHeight, walletAddress],
+    queryFn: async () => {
+      if (!walletAddress || !arweave || !arIOReadSDK) {
+        throw new Error(
+          'Error: Wallet Address, arweave, or arIOReadSDK is not initialized',
+        );
+      }
+
+      const [mioBalance, winstonBalance] = await Promise.all([
```

Comment:
Not a huge deal and I could see the arguments for keeping these in sync, but consider having a useIOBalance hook, and a useARBalance hook, then this hook wraps those. 

That way if it cannot get one balance it will get the other.

Reply:
Note: test what happens with a wallet that doesnt exist yet (has never has an AR balance)

Reply:
Balance showing up as 0 0 as expected with an empty wallet. I like the idea to split this into two hooks but I think it's in good enough shape to release and want to move this story forward. I filed PE-7095 to track this task and will schedule for 1.4.1 or 1.5.0.